Document `name` and `id` (#475)
iterative/terraform-provider-iterative: docs/resources/task.md

-> **Note:** `output` is relative to `workdir`, so `storage { workdir = "foo", output = "bar" }` means "upload `./foo/`, change working directory to the uploaded folder, run `script`, and download `bar` (i.e. `./foo/bar`)".

~> **Warning:** Setting `name` to `Hello, World!` will always produce the `tpi-hello-world-5kz6ldls-57wo7rsp` deterministic identifier. Using the `name` argument is discouraged, and it may be removed or have a different behavior in future versions.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

> - `id` - Task identifier.

Reviewer: ```suggestion
- `id` - Task identifier, `tpi-{name}-{random_hash_1}-{random_hash_2}`. Either the full `{id}` or (if too long), the shorter `{random_hash_1}{random_hash_2}` is used as the name for all cloud resources.
```

- `ssh_public_key` - Used to access the created machines.
- `ssh_private_key` - Used to access the created machines.
- `addresses` - IP addresses of the currently active machines.
- `status` - Status of the machine orchestrator.
- `events` - List of events for the machine orchestrator.
- `logs` - List with task logs; one for each machine.

> ~> **Warning:** `events` doesn't produce a stable output between cloud providers and are intended for human consumption only.

Author: Yes, `status` is stable since #322

> ~> **Warning:** `events` doesn't produce a stable output between cloud providers and are intended for human consumption only.

Reviewer: ```suggestion
~> **Warning:** `events` have different formats across cloud providers and cannot be relied on for programmatic consumption/automation.
```

> 
> -> **Note:** `id` is always in the form `tpi-{name}–{1}-{2}`, where `{name}`, `{1}` and `{2}` are randomly generated identifiers. All the cloud resources are named `id` when legth limits allow it, or `{1}{2}` in cases where there is a more strict length limit.

Reviewer: ```suggestion
```
